serializer.go

```diff
@@ -158,13 +170,47 @@ func (s *Serializer) WriteVariableByteSlice(data []byte, lenType SeriSliceLength
 }
 
 // Write32BytesArraySlice writes a slice of arrays of 32 bytes to the Serializer.
-func (s *Serializer) Write32BytesArraySlice(data SliceOfArraysOf32Bytes, lenType SeriSliceLengthType, errProducer ErrProducer) *Serializer {
+func (s *Serializer) Write32BytesArraySlice(producer WriteSliceOfArraysOf32BytesProducerFunc, deSeriMode DeSerializationMode, lenType SeriSliceLengthType, arrayRules *ArrayRules, errProducer ErrProducer) *Serializer {
 	if s.err != nil {
 		return s
 	}
-	_ = s.writeSliceLength(len(data), lenType, errProducer)
+
+	data := producer()
+	sliceLength := len(data)
+
+	var arrayElementValidator ElementValidationFunc
+	if arrayRules != nil && deSeriMode.HasMode(DeSeriModePerformValidation) {
+		if err := arrayRules.CheckBounds(uint16(sliceLength)); err != nil {
+			s.err = errProducer(err)
+			return s
+		}
+
+		switch arrayRules.ValidationMode {
+		case ArrayValidModeNoValidation:
+		case ArrayValidModeDuplicates:
+			arrayElementValidator = arrayRules.ElementUniqueValidator()
+		case ArrayValidModeLexicalOrdering:
+			arrayElementValidator = arrayRules.LexicalOrderValidator()
+		case ArrayValidModeDuplicates | ArrayValidModeLexicalOrdering:
+			arrayElementValidator = arrayRules.LexicalOrderWithoutDupsValidator()
+		default:
+			panic(ErrUnknownArrayValidationMode)
+		}
+	}
```

Comment:
This piece of code is duplicated at `Write64BytesArraySlice()`. Please make a separate function which given the validation mode inits the ElementValidationFunc appropriately. In fact, the switch statement should be within a method of `ArrayRules`.